sql/plan/ddl.go

```diff
@@ -97,24 +107,38 @@ func (c *CreateTable) RowIter(ctx *sql.Context) (sql.RowIter, error) {
 		if err != nil && !(sql.ErrTableAlreadyExists.Is(err) && c.ifNotExists) {
 			return sql.RowsToRowIter(), err
 		}
-		//TODO: in the event that foreign keys aren't supported, you'll be left with a created table and no foreign keys
-		//this also means that if a foreign key fails, you'll only have the ones declared up to the failure
-		if len(c.fkDefs) > 0 {
+		//TODO: in the event that foreign keys or indexes aren't supported, you'll be left with a created table and no foreign keys/indexes
+		//this also means that if a foreign key or index fails, you'll only have what was declared up to the failure
```

Comment:
we definitely want to fail if part of the table definition can't be satisfied

Reply:
In order to do that, we'd need to either make use of transactions, add a `Close()` on the table creation interface, or have discrete calls for table creation with no fks & idxs, idxs & no fks, etc. Back when adding foreign keys, I discussed this with Zach as I was going to go the `Close()` route, but we decided to re-use the existing interfaces for now, and circle back to this when we add proper transaction support.

Reply:
It does leave you in a wonky state since the table is partially defined, but since it's easily fixed we found it sufficient for now.

Reply:
That makes sense